src/store/mixins/ObservableStoreMixin.ts

```diff
@@ -0,0 +1,693 @@
+				}
+				else {
+					newData.push(update.updates[index]);
+				}
+			});
+
+			update.deletes.sort().reverse().forEach((id) => {
+				if (this.localIndex.has(id)) {
+					newData.splice(this.localIndex.get(id)!, 1);
+				}
+			});
+
+			return newData;
+		}
+
+		/**
+		 * Iterates through the provided items and/or IDs and notifies observers. If items is provided, then the
+		 * observers for that item, and the observers for sets of items that include that are updated. If items is null, then
+		 * these are delete notifications for observers of multiple items. In this case, no update is sent to individual
+		 * observers, and observers of sets receive `ItemUpdate` objects with the IDs of the deleted items and an undefined
+		 * item
+		 *
+		 * @param items Items to send updates for, or null if these are delete notifications for item set observers
+		 * @param ids - IDs of the items, should be in the same order as items
+		 */
+		private notifyItemObservers(items: T[] | null, ids: string[]) {
+			const notify = (id: string, after?: any) => {
+				if (this.itemObservers.has(id)) {
+					this.itemObservers.get(id)!.map((observerOrEntry): Observer<ItemUpdate<T>> | null => {
+						if (isObserverEntry(observerOrEntry)) {
+							return observerOrEntry.observer;
+						}
+						else {
+							return null;
+						}
+					}).filter((observerEntry) => {
+						return observerEntry;
+					}).forEach((observer: Observer<ItemUpdate<T>>) => {
+						observer.next({
+							item: after,
+							id: id
+						});
+					});
+					if (after) {
+						this.itemObservers.get(id)!.map((observerOrEntry): Observer<T> | null => {
+							if (isObserver(observerOrEntry)) {
+								return observerOrEntry;
+							}
+							else {
+								return null;
+							}
+						}).filter((observer) => {
+							return observer;
+						}).forEach((observer: Observer<T>) => {
+							observer.next(after);
+						});
+					}
+				}
+			};
+			if (items) {
+				items.forEach((after: T, index: number) => {
+					const id = ids[index] || this.identify(after);
+					notify(id, after);
+				});
+			}
+			else {
+				ids.forEach((id) => {
+					notify(id, undefined);
+				});
+			}
+		}
+
+		/**
+		 * Queues the appropriate update and then either starts up a fetch or just triggers sending the updates depending
+		 * on the `fetchAroundUpdates` property
+		 * @param updates Updated items
+		 * @param adds Added items
+		 * @param deletes Deleted IDs
+		 */
+		private sendUpdatesOrFetch(updates: T[], adds: T[], deletes: string[]) {
+			const newUpdate = {
+				updates: updates,
+				adds: adds,
+				deletes: deletes,
+				beforeAll: [],
+				afterAll: []
+			};
+			this.queuedUpdate = this.queuedUpdate ? mergeDeltas(this, this.queuedUpdate, newUpdate) : newUpdate;
+			if (this.fetchAroundUpdates) {
+				this.fetchAndSendUpdates(this);
+			}
+			else {
+				this.sendUpdates();
+			}
+		}
+	};
+}
+export class DefaultObservableStore<T> extends
```

Comment:
I think this will get hit by the issue that incorrect code is emitted in the `d.ts`, you'll need to declare the mixed type in a `const` then use it in for `extends`. Something like this:

```ts
export const DefaultObservableStoreBase = ObservableStoreMixin<T, CrudOptions, UpdateResults<T>, typeof StoreBase>(StoreBase)

export class DefaultObservableStore<T> extends DefaultObservableStoreBase<T, ObservableStoreMixinOptions> { }
```